scripts/Muon/GUI/Common/ADSHandler/muon_workspace_wrapper.py

```diff
@@ -60,14 +36,17 @@ class attribute. Either way, a single property "workspace" will always return it
     A basic muon workspace which is either the workspace or the name of the workspace in the ADS
     """
 
-    def __init__(self, workspace, name=''):
+    def __init__(self, workspace):
         self._is_in_ads = False
         self._workspace = None
         self._directory_structure = ""
         self._workspace_name = ""
 
-        self.workspace = workspace
-        self.name = name
+        if isinstance(workspace, Workspace):
+            self.workspace = workspace
+        else:
+            self.name = workspace
```

Comment:
should this be `self._name`?

Reply:
Again I think going via the setter method is correct here